akka-stream/src/main/scala/akka/stream/impl/io/IOSinks.scala

```diff
@@ -66,3 +72,94 @@ private[akka] final class OutputStreamSink(createOutput: () ⇒ OutputStream, va
   override def withAttributes(attr: Attributes): Module =
     new OutputStreamSink(createOutput, attr, amendShape(attr))
 }
+
+/**
+ * INTERNAL API
+ * Creates simple synchronous (Java 6 compatible) Sink which provide all incoming element on read request via
+ * InputStream.
+ */
+private[akka] sealed class InputStreamSink(timeout: FiniteDuration, val attributes: Attributes, shape: SinkShape[ByteString])
```

Comment:
previously commented already that `final` would be nicer here :-)
btw, not really `iso` "because java" :wink: